scripts/update-index-alias.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+"""
+A script to update the search index an alias is assoiciated with.
+
+To delete old indexes (those with no alias), set --delete-old-indexes=yes
+
+Usage:
+scripts/update-index-alias.py <alias> <target> <stage> <search-api-endpoint> [options]
+
+Options:
+    --delete-old-indexes=<delete-old-indexes>  [default: ]
+"""
+import os
+import sys
+import subprocess
+import yaml
+import json
+import requests
+from requests.exceptions import HTTPError
+from docopt import docopt
+
+
+def update_index_alias(alias, target, stage, endpoint, delete_old_indexes):
+    url = "https://{}/{}".format(endpoint, alias)
+    auth_token = _get_auth_token(stage)
+    headers = {
+        'Authorization': "Bearer {}".format(auth_token),
+        'Content-type': 'application/json'
+    }
+    data = {'type': 'alias', 'target': "{}".format(target)}
+
+    response = requests.put(url, headers=headers, json=data)
+
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        print("HTTPError updating alias: {}".format(e.args[0]))
+        sys.exit(1)
+    except Exception as e:
+        print("Error updating alias: {}".format(e))
+        sys.exit(2)
+
+    print('Successfully updated alias')
+
+    if delete_old_indexes == 'yes':
+        _delete_old_indexes(endpoint, auth_token)
+
+
+def _get_auth_token(stage):
+    DM_CREDENTIALS_REPO = os.environ.get('DM_CREDENTIALS_REPO')
+    creds = subprocess.check_output([
+        "{}/sops-wrapper".format(DM_CREDENTIALS_REPO),
+        "-d",
+        "{}/vars/{}.yaml".format(DM_CREDENTIALS_REPO, stage)
+    ])
+    return yaml.load(creds)['search_api']['auth_tokens'][2]
+
+
+def _delete_old_indexes(endpoint, auth_token):
```

Comment:
I'm not very comfortable with it deleting all old indexes, particularly if this will become part of the Jenkins job. I would be far more comfortable with it having an option to delete just the index that has been replaced, although I suppose this is arguably the most useful index to keep.

Should we instead introduce something like a `g-cloud-old` alias? This alias would be applied to the current `g-cloud` aliased index, and then when the script runs, it shuffles indexes back along this pathway. I.e. the new index takes the `g-cloud` alias, the current `g-cloud` index takes the `g-cloud-old` alias, and then the script deletes the index that was covered by the `g-cloud-old` alias.

This would meet the function of cycling out old indexes, but give us the ability to quickly restore the last version of the index in case of mistakes or problems.